backend/app/api/src/helpers/SetupStepsUpdater.ts

```diff
@@ -0,0 +1,194 @@
+import {
+    Organization,
+    OrganizationRegistrationPeriod,
+    Platform,
+} from "@stamhoofd/models";
+import { SetupStepType, SetupSteps } from "@stamhoofd/structures";
+
+export class SetupStepUpdater {
+    private static readonly STEP_TYPE_OPERATIONS: Record<SetupStepType, (setupSteps: SetupSteps,
+        organization: Organization,
+        platform: Platform) => void> = {
+        [SetupStepType.Groups]: this.updateStepGroups,
+        [SetupStepType.Premises]: this.updateStepPremises
+    }
+
+    static async updateSetupStepsForAllOrganizationsInCurrentPeriod({
+        batchSize, stepTypes, platforms
+    }: { batchSize?: number, stepTypes?: SetupStepType[], platforms?: Platform[] } = {}) {
+
+        if(!platforms) {
+            platforms = await Platform.all();
+        }
+
+        for (const platform of platforms) {
+            const periodId = platform.periodId;
+
+            let lastId = "";
+
+            // eslint-disable-next-line no-constant-condition
+            while (true) {
+                const organizationRegistrationPeriods =
+                    await OrganizationRegistrationPeriod.where(
+                        {
+                            id: { sign: ">", value: lastId },
+                            periodId: periodId,
+                        },
+                        { limit: batchSize ?? 10, sort: ["id"] }
+                    );
+
+                if (organizationRegistrationPeriods.length === 0) {
+                    lastId = "";
+                    break;
+                }
+
+                for (const organizationRegistrationPeriod of organizationRegistrationPeriods) {
+                    const organizationId =
+                        organizationRegistrationPeriod.organizationId;
+                    if (!organizationId) {
+                        continue;
+                    }
+                    console.log(
+                        "[FLAG-MOMENT] checking flag moments for " +
+                            organizationId
+                    );
+                    const organization = await Organization.getByID(
```

Comment:
Het zou netter zijn om alle organizations van de hele batch in één keer op te vragen en hier te zoeken, dat zou performanter moeten zijn omdat je maar één query nodig hebt